Remotion/Data/DomainObjects/Persistence/Rdbms/SqlServer/Model/Building/SqlStorageTypeInformationProvider.cs

```diff
@@ -226,6 +229,42 @@ public IStorageTypeInformation GetStorageType (object? value)
       return null;
     }
 
+    private StorageTypeInformation GetDateTimeStorageType (bool isNullableInDatabase, Type dotNetType, IPropertyInformation? propertyInformation)
+    {
+      if (propertyInformation != null)
+      {
+        var storageTypeFromPropertyInfo = GetDateTimeStorageTypeFromPropertyInformation(propertyInformation, isNullableInDatabase, dotNetType);
+
+        if (storageTypeFromPropertyInfo != null)
+          return storageTypeFromPropertyInfo;
+      }
+
+      return new StorageTypeInformation(typeof(DateTime), "datetime2", DbType.DateTime2, isNullableInDatabase, null, dotNetType, new DefaultConverter(dotNetType));
+    }
+    private StorageTypeInformation? GetDateTimeStorageTypeFromPropertyInformation (IPropertyInformation propertyInfo, bool isNullableInDatabase, Type dotNetType)
```

Comment:
```suggestion
    }

    private StorageTypeInformation? GetDateTimeStorageTypeFromPropertyInformation (IPropertyInformation propertyInfo, bool isNullableInDatabase, Type dotNetType)
```

Reply:
After flipping the operations in the GetDateTimeStorateType method, I'm not sure if we shouldn't inline the GetDateTimeStorageTypeFromPropertyInformation method. I can see the code stucturing, on the other hand, the method isn't helpful all alone. probably something one needs to see in full to appriciate.

Reply:
I think we should keep it extra, so it's obvious what happens if we do have a property information

Reply:
okay.